samples/nrf9160/modem_shell/src/ltelc/ltelc.c

```diff
@@ -0,0 +1,515 @@
+		}
+	} break;
+	case LTE_LC_EVT_MODEM_SLEEP_EXIT_PRE_WARNING:
+	case LTE_LC_EVT_MODEM_SLEEP_ENTER:
+	case LTE_LC_EVT_MODEM_SLEEP_EXIT:
+		ltelc_shell_print_modem_sleep_notif(uart_shell, evt);
+		break;
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+		/** The currently active LTE mode is updated. If a system mode that
+		 *  enables both LTE-M and NB-IoT is configured, the modem may change
+		 *  the currently active LTE mode based on the system mode preference
+		 *  and network availability. This event will then indicate which
+		 *  LTE mode is currently used by the modem.
+		 */
+		shell_print(uart_shell, "Currently active system mode: %s",
+			    ltelc_shell_sysmode_currently_active_to_string(
+				    evt->lte_mode, snum));
+		break;
+	case LTE_LC_EVT_NW_REG_STATUS:
+		ltelc_shell_print_reg_status(uart_shell, evt->nw_reg_status);
+
+#if defined(CONFIG_MODEM_INFO)
+		if (evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_EMERGENCY ||
+		    evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ||
+		    evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_ROAMING) {
+			k_work_submit(&modem_info_work);
+		}
+#endif
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		shell_print(uart_shell,
+			    "LTE cell changed: Cell ID: %d, Tracking area: %d",
+			    evt->cell.id, evt->cell.tac);
+		break;
+	case LTE_LC_EVT_RRC_UPDATE:
+		shell_print(uart_shell, "RRC mode: %s",
+			    evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
+			    "Connected" :
+			    "Idle");
+		break;
+	case LTE_LC_EVT_PSM_UPDATE:
+		shell_print(
+			uart_shell,
+			"PSM parameter update: TAU: %d, Active time: %d seconds",
+			evt->psm_cfg.tau, evt->psm_cfg.active_time);
+		break;
+	case LTE_LC_EVT_EDRX_UPDATE: {
+		char log_buf[60];
+		ssize_t len;
+
+		len = snprintf(log_buf, sizeof(log_buf),
+			       "eDRX parameter update: eDRX: %f, PTW: %f",
+			       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
+		if (len > 0) {
+			shell_print(uart_shell, "%s", log_buf);
+		}
+		break;
+	}
+	default:
+		break;
+	}
+}
+
+/******************************************************************************/
+
+static int ltelc_default_pdp_context_set(void)
+{
+	int ret;
+
+	if (ltelc_sett_is_defcont_enabled() == true) {
+		ret = pdn_ctx_configure(0, ltelc_sett_defcont_apn_get(),
+					ltelc_sett_defcont_pdn_family_get(),
+					NULL);
+		if (ret) {
+			printf("pdn_ctx_configure returned err %d", ret);
+			return ret;
+		}
+	}
+	return 0;
+}
+static int ltelc_default_pdp_context_auth_set(void)
+{
+	int ret;
+
+	if (ltelc_sett_is_defcontauth_enabled() == true) {
+		ret = pdn_ctx_auth_set(0, ltelc_sett_defcontauth_prot_get(),
+				       ltelc_sett_defcontauth_username_get(),
+				       ltelc_sett_defcontauth_password_get());
+
+		if (ret) {
+			shell_error(uart_shell,
+				    "pdn_ctx_auth_set returned err  %d", ret);
+			return ret;
+		}
+	}
+	return 0;
+}
+
+static int ltelc_normal_mode_at_cmds_run(void)
+{
+	char *normal_mode_at_cmd;
+	char response[CONFIG_AT_CMD_RESPONSE_MAX_LEN + 1];
```

Comment:
It's 2700 bytes by default (it seems that the sample uses the default value unless I missed the config somewhere), quite a lot for the buffer allocated on a stack.

Reply:
If this is in stack, let's give it a fixed size.

Reply:
I've left these reservations in the stack. There are 4 similar big AT response allocations (and more small ones). I've defined `MOSH_AT_CMD_RESPONSE_MAX_LEN 2700` for this purpose so we are not dependent on outside configuration w.r.t. our stack usage.
I didn't move these outside of the stack as we have rather big stack anyway and this doesn't seem to cause issues.